app/src/pages/home/Search.tsx

```diff
@@ -0,0 +1,95 @@
+import { useMemo, useState } from "react";
+import { FaAngleDown, FaAngleUp } from "react-icons/fa6";
+import { Link } from "react-router-dom";
+import { useAtom } from "jotai";
+import { sortBy, startCase } from "lodash";
+import Badge from "@/components/Badge";
+import Input from "@/components/Input";
+import { flatLessons } from "@/data/data";
+import { languageSearch, titleSearch, topicSearch } from "@/pages/Home";
+import classes from "./Search.module.css";
+
+function Search() {
+  /** page state */
+  const [getTitleSearch, setTitleSearch] = useAtom(titleSearch);
+  const [getTopicSearch, setTopicSearch] = useAtom(topicSearch);
+  const [getLanguageSearch, setLanguageSearch] = useAtom(languageSearch);
+  const [showAll, setShowAll] = useState(false);
+
+  /** filtered search results */
+  const results = useMemo(() => {
+    /** split search text into separate terms */
+    const nameTerms = getTitleSearch.toLowerCase().split(/\s+/);
+    const topicSearch = getTopicSearch.toLowerCase();
+    const languageSearch = getLanguageSearch.toLowerCase();
+
+    /** filter by search terms */
+    const filtered = flatLessons.filter(
+      ({ lesson, title, topic, language }) =>
+        nameTerms.every(
+          (term) => title.includes(term) || lesson.toLowerCase().includes(term),
+        ) &&
+        (topic === "" || topic.toLowerCase().includes(topicSearch)) &&
+        language.toLowerCase().includes(languageSearch),
+    );
+
+    /** sort lowest completion % */
+    return sortBy(filtered, ["completion"]);
```

Comment:
```suggestion
    return filtered;
```

Just throwing this out there, maybe we always preserve the order? For one thing, maybe if a video early in a series is, say, 80% checked, it has all the more reason to be shown near the top, because bringing it to completion should be easier. Also, it's kind of encouraging to see some completions, rather a pile of 0%.

Reply:
Totally up to you for what you want to prioritize. Feel free to commit the suggestion.